Gems/Terrain/Assets/Shaders/Terrain/TerrainSrg.azsli

```diff
@@ -75,6 +77,47 @@ ShaderResourceGroup TerrainSrg : SRG_Terrain
         float2 m_offset; // x/y offset of min x/y corner of grid.
     };
 
+    struct ClipmapData
+    {
+        // The 2D xy-plane view position where the main camera is.
+        // xy: previous; zw: current.
+        float4 m_viewPosition;
+
+        // 2D xy-plane world bounds defined by the terrain.
+        // xy: min; zw: max.
+        float4 m_worldBounds;
+
+        // The max range that the clipmap is covering.
+        float2 m_maxRenderSize;
+
+        // The size of a single clipmap.
+        float2 m_clipmapSize;
+
+        // Clipmap centers in normalized UV coordinates [0, 1].
+        // xy represent previous clipmap centers, zw represent current clipmap centers.
+        // (Array elements will always be padded to 16, a float4 size. Storing both centers in float4 saves bandwidth.)
+        // They are used for toroidal addressing and may move each frame based on the view point movement.
+        // The move distance is scaled differently in each layer.
+        float4 m_clipmapCenters[ClipmapStackSize];
+
+        // A list of reciprocal the clipmap scale [s],
+        // where 1 pixel in the current layer of clipmap represents s meters. 
+        // Fast lookup list to avoid redundant calculation in shaders.
+        float4 m_clipmapScaleInv[ClipmapStackSize];
+    };
+
+    // Clipmap SRG
+    ClipmapData m_clipmapData;
+
+    Texture2DArray<float4> m_macroColorClipmaps;
+    Texture2DArray<float4> m_macroNormalClipmaps;
+    Texture2DArray<float4> m_detailColorClipmaps;
+    Texture2DArray<float4> m_detailNormalClipmaps;
+    Texture2DArray<float> m_detailHeightClipmaps;
+    // Miscellany clipmap combining:
+    // specularF0, metalness, occlusion, height
+    Texture2DArray<float4> m_detailMiscClipmaps;
```

Comment:
There is still alpha channel in the color unused, where I tried to fill some data in there to save bandwidth. But still ended needing more.
Another major concern is sampling 4 times instead of once.
I haven't thought of disabling those channels. We can make a change later if that happens. I have refactored the code to easily expand clipmaps.

Reply:
Another thing to keep on your radar long term is we'll likely want to do gpu texture compression on the fly when writing out clipmaps, which will affect how many channels we pack into a texture as well.